[helm-operator] Add annotation to use helm upgrade force (#2773)
operator-framework/operator-sdk: website/content/en/docs/helm/reference/advanced_features.md

```sh
$ operator-sdk exec-entrypoint helm --max-workers 10
```

## Use `helm upgrade --force` for deployment

> By adding the annotation `helm.sdk.operator-framework.io/upgrade-force: "true"` to the deployed CR, the operator uses the `force` flag of helm to replace the rendered resources.

Reviewer: ```suggestion
By adding the annotation `helm.operator-sdk/upgrade-force: "true"` to the deployed CR, the operator uses the `force` flag of helm to replace the rendered resources. For more information about this Helm option check the [Helm Upgrade documentation](https://helm.sh/docs/helm/helm_upgrade/).
```

```yaml
apiVersion: example.com/v1alpha1
kind: Nginx
metadata:
  name: example-nginx
  annotations:

>     helm.sdk.operator-framework.io/upgrade-force: "true"

Reviewer: ```suggestion
    helm.operator-sdk/upgrade-force: "true"
```

>     helm.sdk.operator-framework.io/upgrade-force: "true"

Reviewer: ```suggestion
   helm.operator-sdk/upgrade-force: "True"
```
